prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitArchive.java

```
@@ -0,0 +1,91 @@
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.eclipse.jgit.api.ArchiveCommand;
+import org.eclipse.jgit.archive.ZipFormat;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.io.FileOutputStream;
+
+@AllArgsConstructor
+public class GitArchive extends BaseWorkFlowTask {
+
+	@Autowired
+	private GitUtils gitUtils;
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(WorkParameter.builder().key(this.gitUtils.getGitRepoPath()).type(WorkParameterType.TEXT)
+				.optional(true).description("path where the git repo is located").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String path = this.gitUtils.getRepoPath(workContext);
+		if (path == null || path == "") {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No valid path: " + path));
+		}
+
+		Repository repo = this.getRepo(path);
+		if (repo == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No repository at " + path));
+		}
+
+		try {
+			var archivePath = this.archive(repo);
+			workContext.put(this.gitUtils.getContextArchivePath(), archivePath.toAbsolutePath().toString());
+		}
+		catch (Exception e) {
+			repo.close();
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("Cannot archive the repository:" + e));
+		}
+
+		repo.close();
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext, null);
+	}
+
+	private Repository getRepo(String path) {
+		Path gitDir = Paths.get(path);
+		try {
+			Repository repository = new FileRepositoryBuilder().setGitDir(gitDir.toFile()).build();
+			return repository;
+		}
+		catch (Exception e) {
```

Why not to throw this exception from this method and handle it in line 54. We would remove bunch of code lines and add handle if null in finally block

---

The main reason is to have a null option and be able to return a human-readable name in there.